lib/fs.js

```diff
@@ -562,8 +562,13 @@ function write(fd, buffer, offset, length, position, callback) {
     return binding.writeBuffer(fd, buffer, offset, length, position, req);
   }
 
-  if (typeof buffer !== 'string')
-    buffer += '';
+  if (typeof buffer !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'buffer',
+      ['string', 'Buffer', 'TypedArray', 'DataView'],
+      buffer
+    );
+  }
```

Comment:
I think maybe it's better to move this kind of validation into `lib/internal/fs/utils.js` or `lib/internal/validators.js` to reduce duplicated code.

Reply:
Fixed, while not being super happy with the function name. I feared that someone might use it in situations that are not suitable.